спринта №17

Gammius opened the pull request.

Gammius committed:
спринта №17

Gammius committed:
спринта №17
исправление критических ошибок
1.Кнопка play отображается верно (добавил отображение)
2.История поиска показывается при фокусе и пустой строке ( поменял условия)
3.тема меняется в SettingsActivity коректно (активити не от того наследовалось)
4.Иконки работают(добавил флаг в интент)

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -4,7 +4,8 @@
+    android:orientation="vertical"
+    tools:context=".presentation.settings.activity.SettingsActivity">
 
     <LinearLayout
```
В качестве заголовка экрана лучше использовать `Toolbar`. Это стандартный компонент. В него можно передать текст заголовка, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.
Для остальных экранов **аналогично**.

Gammius replied:
Привет. уже есть замечание в копилке по рефактарингу xml. В свободное время займусь.
я посмотрел. ты мне это и советовал =)

IliaAnK commented:
Рекомендация к каждой конструкции на экране "Настройки"
```
<LinearLayout> 
    <TextView/>
    <Switch или ImageView/> 
</LinearLayout>
```
Во-первых, это излишняя вложенность, которая негативно сказывается при отрисовке фрейма.
Во-вторых, она вызывает трудности с понимание того как свёрстан экран;
В-третьих, проще формировать и применять стили.

Каждая такая конструкция может быть представлена одной View.
Для свичера:
```
<com.google.android.material.switchmaterial.SwitchMaterial
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:gravity="center_vertical"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/black_theme_settings"
    android:textSize="16sp" />
```
Для TextView с иконкой справа:
```
<com.google.android.material.textview.MaterialTextView
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:drawableEnd="@drawable/icon_round"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/share_app_settings"
    android:textSize="16sp" />
```

Общие характеристика вынести само собой можно вынести в стили.

Итого иерархия всего `layout`-а будет вот такой:
```
<LinearLayout>
    <MaterialToolbar/>
    <SwitchMaterial/>
    <MaterialTextView/>
    <MaterialTextView/>
    <MaterialTextView/>
<LinearLayout/>
```

Gammius committed:
спринта №17
исправление критических ошибок
Отображение контейнера с историей поиска добавил еще одно условие что история должна быть не пуста
+ баг:при нажатиие на очистить поиск отображался контейнер с историей поиска (если пустой)
+ баг: при нажатие на кнопку очистить историю отображался результат предыдушего поиска

Теперь все критерии выполнаются:
Пользователь находится на экране «Поиск», поле поискового запроса пустое и находится в фокусе. При условии, что пользователь имеет непустую историю и не очищал историю поиска, он видит:
Заголовок «История поиска».
Список просмотренных ранее треков.
Кнопку «Очистить историю».
